src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -332,20 +337,22 @@ internal Location CreateLocation(LocationVersionOne v1Location)
                 {
                     FullyQualifiedLogicalName = v1Location.LogicalLocationKey ?? v1Location.FullyQualifiedLogicalName,
```

Comment:
>v1Location.LogicalLocationKey [](start = 48, length = 29)

I don't think this was ever right. `LogicalLocationKey` was a synthetic key, like `N1::N2-1`, that was used when two FQLNs were identical. It would never have been right to assign that value to a v2 FQLN.

In fact, this logic is backwards. It should be:
```
FullyQualifiedLogicalName = v1Location.FullyQualifiedLogicalName ?? v1Location.LogicalLocationKey,
```
v1 FQLN is guaranteed to be the right answer if it's there; if it's absent, it's because it matched the key.

Reply:
[https://github.com/Microsoft/sarif-sdk/issues/1182](https://github.com/Microsoft/sarif-sdk/issues/1182) marked urgent

---
In reply to: [243868044](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243868044) [](ancestors = 243868044)